src/main/java/io/stargate/sgv2/jsonapi/exception/mappers/ThrowableToErrorMapper.java

```diff
@@ -47,11 +57,30 @@ public final class ThrowableToErrorMapper {
                 message, fieldsForMetricsTag, fields, Response.Status.GATEWAY_TIMEOUT);
           }
         }
-        // add error code as error field
-        Map<String, Object> fields =
-            debugEnabled ? Map.of("exceptionClass", throwable.getClass().getSimpleName()) : null;
-        Map<String, Object> fieldsForMetricsTag =
-            Map.of("exceptionClass", throwable.getClass().getSimpleName());
+        if (throwable instanceof UnauthorizedException
+            || throwable
+                instanceof com.datastax.oss.driver.api.core.servererrors.UnauthorizedException) {
+          return new CommandResult.Error(
+              "UNAUTHENTICATED: Invalid token",
+              fieldsForMetricsTag,
+              fields,
+              Response.Status.UNAUTHORIZED);
+        } else if (throwable instanceof QueryValidationException) {
+          if (message.contains("vector<float,")) { // TODO is there a better way?
```

Comment:
The driver throw it as QueryValidationException in case of vector dimension mismatch. This exception is used for many other errors., not sure if there is other way to identify it.

Reply:
Yes, also hard to know the exact table dimension when having an error.
If 2, but got 5,  it will be "Unexpected 12 extraneous bytes after vector<float, 2> value'
If 5, but got 2,  it will be "Not enough bytes to read a vector<float, 5> '